module/x/peggy/keeper/attestation.go

```diff
@@ -84,31 +84,34 @@ func (k Keeper) TryAttestation(ctx sdk.Context, att *types.Attestation) {
 }
 
 // processAttestation actually applies the attestation to the consensus state
-func (k Keeper) processAttestation(ctx sdk.Context, att *types.Attestation, claim types.EthereumClaim) {
+func (k Keeper) processAttestation(ctx sdk.Context, att *types.Attestation, claim types.EthereumClaim) error {
 	lastEventNonce := k.GetLastObservedEventNonce(ctx)
 	// this check is performed at the next level up so this should never panic
 	// outside of programmer error.
 	if claim.GetEventNonce() != uint64(lastEventNonce)+1 {
-		panic("attempting to apply events to state out of order")
+		fmt.Errorf("attempting to apply events to state out of order")
```

Comment:
So this panic actually should panic. As long as all `processAttestation` returned errors get panic'ed thats cool too. We really don't want to apply the state transitions out of order